stdlib/public/Concurrency/AsyncLet.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+@_implementationOnly import _SwiftConcurrencyShims
+
+// ==== Async Let -------------------------------------------------------------
+// Only has internal / builtin functions as it is not really accessible directly
+
+//@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+//@_silgen_name("swift_asyncLet_create")
+//func _asyncLetCreate(task: Builtin.NativeObject) -> Builtin.RawPointer
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+@_silgen_name("swift_asyncLet_start")
+public func _asyncLetStart<T>(
+  asyncLet: Builtin.RawPointer,
+  operation: __owned @Sendable @escaping () async throws -> T
```

Comment:
The closure is actually not escaping. So the "@escaping" can be removed

Reply:
Though, we'll need to make sure that the closed over values (the partial_apply arguments) are kept alive until the ``_asyncLetEnd``.